tests/scripts/all.sh

```diff
@@ -3448,6 +3448,35 @@ component_test_malloc_0_null () {
     tests/ssl-opt.sh -e 'proxy'
 }
 
+support_test_aesni () {
+    # require an x64_64 target
+    gcc -v 2>&1 | grep Target | grep -q x86_64
+}
+
+component_test_aesni () { # ~ 20s
+    msg "build: default config with MBEDTLS_HAVE_ASM and MBEDTLS_AESNI_C enabled"
+    scripts/config.py set MBEDTLS_AESNI_C
+    scripts/config.py set MBEDTLS_HAVE_ASM
+
+    msg "AES tests, MBEDTLS_AESNI_HAVE_CODE=1 (asm)"
+    make lib tests CC=gcc CFLAGS='-O2 -Werror -Wall -Wextra -DMBEDTLS_AESNI_HAVE_CODE=1'
+    cd tests
+    for t in `find . -type f -executable -name '*aes*'`; do
+        # Run all the suites with aes in their name
+        ./$t
+    done
+    cd ..
```

Comment:
Please don't complicate things just to save a few seconds of runtime. Running the tests takes less long than the build.
```suggestion
    make test
```